platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -98,63 +106,50 @@
+    Args.notEmpty(coreName, "Cannot create Solr client. Missing core name.");
+    if (getSolrDir() != null) {
+      ConfigurationStore.getInstance().setDataDirectoryPath(getSolrDir());
     }
-    LOGGER.debug("Solr({}): Creating an HTTP Solr client using url [{}]", core, coreUrl);
-    return new SolrClientAdapter(core, () -> createSolrHttpClient(solrUrl, core, coreUrl));
+    LOGGER.debug(
+        "Solr({}): Creating an HTTP Solr client using url [{}]", coreName, getCoreUrl(coreName));
+    return new SolrClientAdapter(coreName, () -> createSolrHttpClient(coreName));
   }
 
   @VisibleForTesting
-  SolrClient createSolrHttpClient(String url, String coreName, String coreUrl)
-      throws IOException, SolrServerException {
-    final HttpClientBuilder builder =
-        HttpClients.custom()
-            .setDefaultCookieStore(new BasicCookieStore())
-            .setMaxConnTotal(128)
-            .setMaxConnPerRoute(32);
+  SolrClient createSolrHttpClient(String coreName) throws IOException, SolrServerException {
+    final HttpClientBuilder httpClientBuilder = createHttpBuilder();
+    final HttpSolrClient.Builder solrClientBuilder =
+        new HttpSolrClient.Builder(getCoreUrl(coreName));
 
-    if (StringUtils.startsWithIgnoreCase(url, "https")) {
-      builder.setSSLSocketFactory(
-          new SSLConnectionSocketFactory(
-              getSslContext(),
-              getProtocols(),
-              getCipherSuites(),
-              SSLConnectionSocketFactory.BROWSER_COMPATIBLE_HOSTNAME_VERIFIER));
-    }
-    createSolrCore(url, coreName, null, builder.build());
     try (final Closer closer = new Closer()) {
-      final HttpSolrClient noRetryClient =
-          closer.with(new HttpSolrClient.Builder(coreUrl).withHttpClient(builder.build()).build());
-      final HttpSolrClient retryClient =
-          closer.with(
-              new HttpSolrClient.Builder(coreUrl)
-                  .withHttpClient(
-                      builder.setRetryHandler(new SolrHttpRequestRetryHandler(coreName)).build())
-                  .build());
-
-      return closer.returning(new PingAwareSolrClientProxy(retryClient, noRetryClient));
+
+      createSolrCore(coreName, null, httpClientBuilder.build());
+
+      CloseableHttpClient noRetryHttpClient = httpClientBuilder.build();
+      final HttpSolrClient noRetrySolrClient =
+          closer.with(solrClientBuilder.withHttpClient(noRetryHttpClient).build());
+
+      CloseableHttpClient yesRetryHttpClient =
+          httpClientBuilder.setRetryHandler(new SolrHttpRequestRetryHandler(coreName)).build();
+      final HttpSolrClient retrySolrClient =
+          closer.with(solrClientBuilder.withHttpClient(yesRetryHttpClient).build());
+
+      return closer.returning(new PingAwareSolrClientProxy(retrySolrClient, noRetrySolrClient));
     }
   }
 
+  private boolean useTls() {
+    return StringUtils.startsWithIgnoreCase(getSolrUrl(), "https");
```

Comment:
❗️ Let's try to avoid retrieving the solr url too many times.

Reply:
Definitely cache and reuse this URL.